src/mongocrypt-ctx-rewrap-many-datakey.c

```diff
@@ -136,6 +136,17 @@ static mongocrypt_kms_ctx_t *_next_kms_ctx_encrypt(mongocrypt_ctx_t *ctx) {
     mongocrypt_ctx_t *dkctx = NULL;
 
     BSON_ASSERT_PARAM(ctx);
+    /* Check if any need retry */
+    {
+        _mongocrypt_ctx_rmd_datakey_t *it = rmdctx->datakeys;
+        while (it != NULL) {
+            _mongocrypt_ctx_datakey_t *dkctx = (_mongocrypt_ctx_datakey_t *)it->dkctx;
+            if (dkctx->kms.should_retry) {
```

Comment:
```suggestion
            if (dkctx->kms.should_retry) {
               dkctx->kms.should_retry = false; // Reset retry state.
```

Similar to other contexts ([example](https://github.com/mongodb/libmongocrypt/blob/a41401895bdde90c8148fc0ef33dbeea87d47d9d/src/mongocrypt-ctx-datakey.c#L46)), reset `should_retry` before returning.